# Hypothesis Testing

# Task 1: State the Hypotheses statement
Based on the problem statement and the objective to investigate whether the observed weekly operating costs are higher than the theoretical model, we can state the hypotheses as follows:

Null Hypothesis (H₀): 𝑊 ≤ 1000 + 5𝑋
Alternative Hypothesis (H₁): 𝑊 > 1000 + 5𝑋

# Task 2: Calculate the Test Statistic:

Use the following formula to calculate the test statistic (t):
where:
1. ˉxˉ = sample mean weekly cost (Rs. 3,050)
2. μ = theoretical mean weekly cost according to the cost model (W = $1,000 + $5X for X = 600 units)
3. σ = 5*25 units
4. n = sample size (25 restaurants)

In [1]:
import math

# Given data
sample_mean = 3050  # sample mean weekly cost in Rs.
theoretical_mean = 1000 + 5 * 600  # theoretical mean weekly cost in Rs.
theoretical_std_dev = 5 * 25  # theoretical standard deviation of weekly costs in Rs.
n = 25  # sample size

# Calculate the test statistic (t-statistic)
t_statistic = (sample_mean - theoretical_mean) / (theoretical_std_dev / math.sqrt(n))

# Print the results
print(f"Sample Mean (x): {sample_mean}")
print(f"Theoretical Mean (μ): {theoretical_mean}")
print(f"Theoretical Standard Deviation (σ): {theoretical_std_dev}")
print(f"Sample Size (n): {n}")
print(f"Test Statistic (t): {t_statistic}")

Sample Mean (x): 3050
Theoretical Mean (μ): 4000
Theoretical Standard Deviation (σ): 125
Sample Size (n): 25
Test Statistic (t): -38.0


# Task 3: Determine the Critical Value:

Using the alpha level of 5% (α = 0.05), determine the critical value from the standard normal (Z) distribution table.

In [3]:
from scipy.stats import norm

# Alpha level (significance level)
alpha = 0.05

# Find the critical value (Z-score) for one-tailed test (right-tailed)
critical_value = norm.ppf(1 - alpha)

# Print the critical value
print(f"Critical Value (Z): {critical_value:.2f}")

Critical Value (Z): 1.64


# Task 4: Make a Decision
Compare the test statistic with the critical value to decide whether to reject the null hypothesis.

To decide whether to reject the null hypothesis based on the test statistic and critical value provided:
1. Test Statistic: 𝑡 = −38.0
2. Critical Value: 𝑍 = 1.64

1. Null Hypothesis (H₀): The null hypothesis typically states that there is no significant difference between the observed 
    sample mean and the theoretical mean.
2. Decision Rule: For a two-tailed hypothesis test (testing if the mean is significantly different from the theoretical mean), 
    we compare the absolute value of the test statistic to the critical value.
3. Comparison: Since the test statistic 𝑡 = −38.0 is much smaller (in absolute terms) than the critical value 𝑍 = 1.64, we can 
    conclude that ∣−38.0∣ > 1.64.
4. Decision: Based on this comparison, ∣−38.0∣ > 1.64, which means the test statistic falls far outside the range of values that 
    would be expected under the null hypothesis. Therefore, we reject the null hypothesis (H₀).

##### Conclusion:

Reject the Null Hypothesis: The test statistic of -38.0 is significantly lower than the critical value of 1.64. This suggests strong evidence against the null hypothesis and indicates that the observed mean weekly operating costs are significantly different (lower in this case) than predicted by the theoretical model.

In summary, based on the test statistic and critical value, we reject the null hypothesis at the chosen significance level, 
implying that there is a significant difference between the observed data and the expected values under the null hypothesis.

# Task 5: Conclusion
Based on the decision in step 4, conclude whether there is strong evidence to support the restaurant owners' claim that the weekly operating costs are higher than the model suggests.

Based on the statistical analysis and the comparison between the test statistic and critical value, there is strong evidence to 
support the restaurant owners' claim that the weekly operating costs are higher than the model predicts. The test statistic of 
-38.0 indicates a significant deviation from the expected costs under the null hypothesis, leading to the rejection of the null 
hypothesis. Therefore, the observed data suggest that the actual weekly operating costs are indeed higher than those predicted 
by the theoretical model.

# Chi-square Test

# Task 1: State the Hypotheses

To determine if there is a significant association between the type of smart home device purchased (Smart Thermostats vs. Smart 
Lights) and the customer satisfaction level, we can use a Chi-Square test for independence.

Null Hypothesis (H₀):
There is no association between the type of smart home device purchased and the customer satisfaction level.

Alternative Hypothesis (H₁):
There is an association between the type of smart home device purchased and the customer satisfaction level.

# Task 2: Compute the Chi-Square Statistic

In [6]:
import numpy as np

# Define the observed contingency table
observed = np.array([
    [50, 70],
    [80, 100],
    [60, 90],
    [30, 50],
    [20, 50]
])

# Compute row and column totals
row_totals = np.sum(observed, axis=1)
col_totals = np.sum(observed, axis=0)
total = np.sum(observed)

# Compute expected frequencies under independence assumption
expected = np.outer(row_totals, col_totals) / total

# Compute the Chi-Square statistic
chi2 = np.sum((observed - expected)**2 / expected)

# Degrees of freedom calculation
# degrees_of_freedom = (number of rows - 1) * (number of columns - 1)
rows, cols = observed.shape
dof = (rows - 1) * (cols - 1)

# Print the results
print("Chi-Square Statistic:", chi2)
print("Degrees of Freedom:", dof)
print("Expected Frequencies:\n", expected)

Chi-Square Statistic: 5.638227513227513
Degrees of Freedom: 4
Expected Frequencies:
 [[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]


# Task 3: Determine the Critical Value

Using the significance level (alpha) of 0.05 and the degrees of freedom (which is the number of categories minus 1)

In [8]:
import scipy.stats as stats

# Given significance level (alpha) and degrees of freedom (df)
alpha = 0.05
df = dof  # Degrees of freedom calculated earlier

# Determine the critical value from the Chi-Square distribution
critical_value = stats.chi2.ppf(1 - alpha, df)

# Print the critical value
print("Critical Value:", critical_value)

Critical Value: 9.487729036781154


# Task 4: Make a Decision
Compare the Chi-Square statistic with the critical value to decide whether to reject the null hypothesis.

From our calculations:
1. Chi-Square statistic: 5.638227513227513
2. Critical value: 9.487729036781154

#### Since 5.638227513227513 ≤ 9.487729036781154:

##### Decision:
The Chi-Square statistic does not exceed the critical value. Therefore, we fail to reject the null hypothesis at the 0.05 significance level.

##### Conclusion:

Based on the Chi-Square test results, there is not enough evidence to conclude that there is a significant association between 
the type of smart home device purchased (Smart Thermostats vs. Smart Lights) and the customer satisfaction level.